hardhat.config.ts

```diff
@@ -0,0 +1,81 @@
+import fs from 'fs';
+import path from 'path';
+
+import '@nomiclabs/hardhat-waffle';
+import '@nomiclabs/hardhat-ethers';
+import '@nomiclabs/hardhat-etherscan';
+
+import 'solidity-coverage';
+import 'hardhat-contract-sizer';
+import 'hardhat-abi-exporter';
+import 'hardhat-gas-reporter';
+import '@typechain/hardhat';
+
+const configPath = path.join(__dirname, '/config.json');
+const configFile = fs.existsSync(configPath) ? JSON.parse(fs.readFileSync(configPath, 'utf8')) : {};
+
+const loadAPIKey = (apiKeyName: string) => {
+    return configFile.apiKeys ? (configFile.apiKeys[apiKeyName] ? configFile.apiKeys[apiKeyName] : '') : '';
+};
+
+// Casting to unknown assume the good type is provided
+const loadENVKey = <T>(envKeyName: string) => {
+    return process.env[envKeyName] as unknown as T;
+};
+
+const configNetworks = configFile.networks || {};
+
+import { HardhatUserConfig } from 'hardhat/config';
+
+const config: HardhatUserConfig = {
+    networks: {
+        hardhat: {
+            gasPrice: 20000000000,
+            gas: 9500000,
+            accounts: {
+                count: 10,
+                accountsBalance: '10000000000000000000000000000'
+            }
+        },
+
+        ...configNetworks
+    },
+
+    solidity: {
+        version: '0.6.12',
```

Comment:
Perhaps we should read this during runtime from package.json, i.e., something like:
```
JSON.parse(fs.readFileSync('package.json')).devDependencies.solc
```

Reply:
I think it's overkill. We'd still have many other locations (solhint, pragmas, etc.) with an explicit version specified

Reply:
Of course, first and foremost - the contracts themselves.
But at least we'll have this dependency slightly reduced.
I think it's worth the cost, unless it yields a performance hit upon starting the tests.

Reply:
It's not standard. BTW, keep in mind that the only reason we're installing `solc` in `package.json` is for solhint. Hardhat (and truffle, previously) download their own version of `solc`. So your suggestion might break things :(